src/core_plugins/kibana/public/management/sections/indices/create_index_pattern_wizard/components/step_time_field/step_time_field.js

```diff
@@ -52,7 +53,20 @@ export class StepTimeField extends Component {
   }
 
   onTimeFieldChanged = (e) => {
-    this.setState({ selectedTimeField: e.target.value });
+    const value = e.target.value;
+
+    // Ensure it exists in the options, if not, make this undefined
+    // This supports the `I do not want to use a time filter` option
+    const foundByFieldName = this.state.timeFields.find(timeField => timeField.fieldName === value);
+
+    // If the value is an empty string, it's not a valid selection
+    const invalidSelection = value === '';
+
+    if (!foundByFieldName) {
+      this.setState({ selectedTimeField: undefined, timeFieldSet: !invalidSelection });
+    } else {
+      this.setState({ selectedTimeField: value, timeFieldSet: !invalidSelection });
+    }
```

Comment:
This seems overly complicated to me.  Couldn't you just always set the selectedTimeField to value?  It is undefined if they chose I do not want, and timeFieldSet is set to true.  Maybe I'm missing something here.